src/test/kotlin/org/rust/ide/refactoring/RsIntroduceVariableHandlerTest.kt

```diff
@@ -315,6 +376,32 @@ class RsIntroduceVariableHandlerTest : RsTestBase() {
         }
     """)
 
+    fun `test lambda has no braces 2`() = doTest("""
```

Comment:
In 1 the extracted variable is the parameter of the closure, so it doesn't exist outside of it. In 2 it is some externally defined variable, so in principle the binding could be placed outside of the closure. It's still preferable to place the binding inside the closure, though, both because we'd like the binding scope to be as small as possible and because placing the `let` outside of the closure would change environment binding mode.

Maybe we should also have a couple of more explicit tests, where the variable with the same name (but semantically different) is defined outside of the closure.

Reply:
Thanks, it is reasonable